src/app/Library/CrudPanel/Traits/Validation.php

```diff
@@ -191,8 +191,7 @@ public function setRequiredFields($classOrRulesArray)
                 }
             }
         }
-
-        $this->setOperationSetting('requiredFields', $requiredFields);
+        $this->setOperationSetting('requiredFields', array_unique(array_merge($this->getOperationSetting('requiredFields') ?? [], $requiredFields)));
```

Comment:
Ugh... such a long line. Let's split it up so it's easy to understand what's happening. Which is also a sine that this needs a comment, explaining why this is needed.

```suggestion
        $requiredFields = array_merge($this->getOperationSetting('requiredFields') ?? [], $requiredFields);
        $requiredFields = array_unique($requiredFields);

        $this->setOperationSetting('requiredFields', $requiredFields);
```